### DISCUSSION TITLE
Change `InputMap.action_get_events(action)` return value from Array to Dictionary

### DISCUSSION THREAD

**Discussion by alvinseptiano** [godotengine/godot-proposals#9539] [Scripting]
Make it easier to extract value from `InputMap.action_get_events()` to determine whether InputEvent is a Joypad or Key.

For example this is the current return value:
[InputEventKey: keycode=87 (W), mods=none, physical=true, pressed=false, echo=false, InputEventJoypadMotion: axis=1, axis_value=-1.00]

After:
{ 
	InputEventKey: [keycode=87 (W), mods=none, physical=true, pressed=false, echo=false], 
	InputEventJoypadMotion: [axis=1, axis_value=-1.00]
	InputEventJoypadButton: null
}
or any similar way so that everything is easier to get and more direct.

**Comment by AThousandShips** [comment]
This would make iterating over all the events not caring about what types they are much more inconvenient, but you can easily just iterate over it and pick what events you want, also this would break compatibility and hurt usefulness as they return a typed array making things easier and safer type wise

**Comment by alvinseptiano** [comment]
Or maybe this can be done by separating `action_get_events()` into each of their own like `InputMap.action_get_mouse_button_event()` or `InputMap.action_get_joypad_button_event()`

**Reply by alvinseptiano** [reply]
I feel like accessing Event more direct like this is more convenient. I don't know 🤷.
